### PR TITLE
Add support for ExternalWorkloads in endpoint profiles

### PR DESCRIPTION
**Pull request by adleong**
When a meshed client attempts to establish a connection directly to the workload IP of an ExternalWorkload, the destination controller should return an endpoint profile for that ExternalWorkload with a single endpoint and the metadata associated with that ExternalWorkload including:
* mesh TLS identity
* workload metric labels
* opaque / protocol hints

### CODE REVIEW

File: controller/api/destination/test_util.go [hunk 1]
```diff
@@ -445,6 +445,25 @@ spec:
...
+apiVersion: workload.linkerd.io/v1alpha1
+kind: ExternalWorkload
+metadata:
+  name: my-cool-workload
+  namespace: ns
+spec:
+  meshTls:
+    identity: spiffe://some-domain/cool
+    serverName: server.local
+  workloadIps:
```
Comment (zaharidichev) [inline controller/api/destination/test_util.go:458]: workloadIps -> workloadIPs

That should fix the test

File: controller/api/destination/watcher/workload_watcher.go [hunk 2]
```diff
@@ -0,0 +1,840 @@
...
+			ip:                 ip,
+			port:               port,
+			pod:                pod,
+			externalWorkload:   externalWorkload,
+			metrics: ipPortVecs.newMetrics(prometheus.Labels{
+				"ip":   ip,
+				"port": strconv.FormatUint(uint64(port), 10),
+			}),
+			log: ww.log.WithFields(logging.Fields{
+				"component": "pod-publisher",
```
Comment (zaharidichev) [inline controller/api/destination/watcher/workload_watcher.go:422]: This should be workload-publisher

**Review by zaharidichev — COMMENTED** [review]
Just some small comments. Overall it looks good to me

**Review by mateiidavid — APPROVED** [review]

**Review by zaharidichev — APPROVED** [review]
LGTM

**Pull request merged**